op_crates/web/00_webidl.js

```diff
@@ -662,10 +662,40 @@
     };
   }
 
-  window.__bootstrap = window.__bootstrap || {};
+  function createPrimitiveEnum(name, opts) {
+    const Enum = new Set(opts.enumValues);
+
+    let defaultValue;
+
+    const hasDefault = "defaultValue" in opts;
+
+    if (hasDefault) {
+      defaultValue = opts.defaultValue;
+    }
+
+    const { converter } = ops;
+
+    return (V, opts = {}) => {
+      V = converter(V);
+
+      if (Enum.has(V)) {
+        return V;
+      } else if (hasDefault) {
+        return defaultValue;
+      } else {
+        throw makeException(
+          TypeError,
+          `The provided value '${V}' is not a valid enum value of type ${name}.`,
+        );
+      }
```

Comment:
This logic doesn't follow spec: https://heycam.github.io/webidl/#es-enumeration.

The default should only be used if V is undefined or null.

Reply:
But really this shouldn't happen in this function at all (see spec), but instead there should be a separate wrapper for converters that make them nullable with a default value.

Reply:
See https://heycam.github.io/webidl/#idl-nullable-type

Reply:
Oh, I didn't catch myself there with the default value, that is *bad*. Passing in an invalid value would instead become a default value, meaning the last else in the chain is unreachable.

You can tell that this was some ad-hoc code I wrote, I needed to review the spec later to get a proper implementation later; for one, I wasn't sure what string types were used in enums, and am suprised to see that the spec asks implementations to work in ECMAScript strings instead of WebIDL strings.